builtin/credential/cert/path_login.go

```diff
@@ -135,7 +152,7 @@ func (b *backend) pathLoginRenew(
 
 		clientCerts := req.Connection.ConnState.PeerCertificates
 		if len(clientCerts) == 0 {
-			return nil, fmt.Errorf("no client certificate found")
+			return logical.ErrorResponse("no client certificate found"), nil
```

Comment:
Does this need a typed error return, maybe `logical.ErrInvalidRequest`?

Reply:
ErrInvalidRequest is actually the default for a logical.ErrorResponse with no typed error.

Reply:
Good to know!